### PR TITLE
some changes for ctrl + c protection

### PR DESCRIPTION
**Pull request by BogdanCiocea**
# Description

Made the case for ctrl + c protection in the todo comment

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -77,7 +77,11 @@ fn main() -> Result<()> {
 
     let ctrlc = Arc::new(AtomicBool::new(false));
     // TODO: make this conditional in the future
-    ctrlc_protection(&mut engine_state, &ctrlc);
+    // This block sets up Ctrl-C signal handling to safely handle user interrupts.
+    // Consider making this configurable in future versions.
+    if cfg!(feature = "ctrlc-protection") {
```
Comment (fdncred) [inline src/main.rs:82]: I don't see this featrure in nushell. What am I missing?

Comment (BogdanCiocea) [reply src/main.rs:82]: It was a TODO in main.rs. I thought it must be done in some way.

Comment (fdncred) [reply src/main.rs:82]: I don't see this feature so landing this PR would effectively disable ctrlc_protection right now.

**Comment by sholderbach** [comment]
Thanks for trying to pick up a todo but I think making this compile time disableable is not a goal here. The commment may allude to the idea that one may want to ignore it under certain runtime circumstances but this would require a much broader look at where this would be the right or wrong thing to do.

**Pull request closed without merging**